ocl: Fix K2x race condition during DSP firmware booting

- Race condition during booting of K2x DSP monitor firmware, causes
  some DSPs not able to create RPMessage object that communicates with
  the host (Arm A15).
- The root cause is that one global data in ipc/RPMessage,
  "RPMessage_module", is placed in global DDR section shared by all cores.
- The fix is to place "ipc/RPMessage" global data into core-private memory.
- MCT-1279

Upstream-Status: Submitted

Signed-off-by: Yuan Zhao <yuanzhao@ti.com>
---
 monitor/cmds/monitor.cmd | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/monitor/cmds/monitor.cmd b/monitor/cmds/monitor.cmd
index a6a9fb9..6ec4151 100644
--- a/monitor/cmds/monitor.cmd
+++ b/monitor/cmds/monitor.cmd
@@ -164,6 +164,13 @@ SECTIONS
     } run=DDR3_VIRT, load=DDR3_PER_CORE
      */
 
+    /* ipc (e.g. RPMessage_module) need to be private for each core */
+    .otherlocalfar :
+    {
+                "*.ae66"            (.fardata)
+                "*.ae66"            (.far)
+    } run=DDR3_VIRT, load=DDR3_PER_CORE
+
     .localddr: run=DDR3_VIRT, load=DDR3_PER_CORE
     .ctools_core_private: run=DDR3_VIRT, load=DDR3_PER_CORE
 
-- 
2.17.1

